src/Serializer/AbstractSerializer.php

```diff
@@ -241,7 +241,14 @@ protected function serializeValue($value)
         }
 
         if (\is_object($value)) {
-            return 'Object ' . \get_class($value);
+            $objectId = null;
+            if (isset($value->id)) {
+                $objectId = $value->id;
```

Comment:
I believe isset() returns TRUE for a protected property, and then an Error will be thrown when you try to access it.

Reply:
Ah looks like I am wrong here, isset returns FALSE for inaccessible property.

Reply:
This also invokes magic `__isset()` (and then `__get()` ) method for objects, which can be problematic if these magic methods have side effects. Which is exactly what I had to investigate and deal with in our own codebase.

I'd propose that this behavior (magically getting some kind of ID of objects present within traceback) should check if the property _actually_ exists _(via `property_exists()`, which doesn't invoke magic methods)_.

Or at least this magic feature should be optional - i,e. there should be an option to disable it.

Reply:
Sorry you ran into issues because of this!

The approach via `property_exists` sounds reasonable to me. Mind opening a dedicated issue or PR for this? Thanks!

Reply:
https://github.com/getsentry/sentry-php/issues/1476